browser/ui/views/brave_actions/brave_shields_action_view.cc

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/views/brave_actions/brave_shields_action_view.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "brave/browser/ui/brave_actions/brave_action_icon_with_badge_image_source.h"  // NOLINT
+#include "chrome/browser/ui/views/chrome_layout_provider.h"
+#include "chrome/browser/ui/views/toolbar/toolbar_ink_drop_util.h"
+#include "components/grit/brave_components_resources.h"
+#include "components/grit/brave_components_strings.h"
+#include "extensions/common/constants.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/models/image_model.h"
+#include "ui/base/resource/resource_bundle.h"
+#include "ui/gfx/geometry/rect.h"
+#include "ui/gfx/geometry/skia_conversions.h"
+#include "ui/views/animation/ink_drop_impl.h"
+#include "ui/views/controls/button/label_button_border.h"
+#include "ui/views/controls/highlight_path_generator.h"
+#include "ui/views/view.h"
+
+namespace {
+class BraveShieldsActionViewHighlightPathGenerator
+    : public views::HighlightPathGenerator {
+ public:
+  BraveShieldsActionViewHighlightPathGenerator() = default;
+
+  // HighlightPathGenerator
+  SkPath GetHighlightPath(const views::View* view) override {
+    return static_cast<const BraveShieldsActionView*>(view)->GetHighlightPath();
+  }
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(BraveShieldsActionViewHighlightPathGenerator);
```

Comment:
Oops, I missed this. `DISALLOW_COPY_AND_ASSIGN` is deprecated.

Reply:
fixed